lib/checkstl.cpp

```diff
@@ -536,6 +487,63 @@ void CheckStl::iterators()
     }
 }
 
+bool CheckStl::compareIteratorAgainstDifferentContainer(const Token* operatorTok, const Token* containerTok, const unsigned int iteratorId, const std::map<unsigned int, const Token*>& iteratorScopeBeginInfo)
+{
+    if (!containerTok)
+        return false;
+    const Token *otherOperand = nullptr;
+    OperandPosition operandPosition = OperandPosition::None;
+    getOperandData(operatorTok, iteratorId, otherOperand, operandPosition);
+
+    if (otherOperand)
```

Comment:
either "inline" getOperandData() above. Or put an early return here.

    if (!otherOperand)
        return false;

Reply:
@danmar I think I got your point, please, look at this commit
 (https://github.com/danmar/cppcheck/pull/1380/commits/5ea3576a6b0244b78f21af5c93aff65f86db7697) I'm so sorry for my miscommunication.